app/src/sharedTest/java/org/oppia/android/app/classroom/ClassroomListFragmentTest.kt

```diff
@@ -0,0 +1,878 @@
+    fakeOppiaClock.setFakeTimeMode(FakeOppiaClock.FakeTimeMode.MODE_UPTIME_MILLIS)
+    storyProgressTestHelper.markCompletedFractionsTopic(
+      profileId = profileId,
+      timestampOlderThanOneWeek = false
+    )
+    logIntoAdminTwice()
+
+    composeRule.onNodeWithTag(PROMOTED_STORY_LIST_HEADER_TEST_TAG).onChildAt(0)
+      .assertTextContains(context.getString(R.string.recommended_stories))
+      .assertIsDisplayed()
+
+    composeRule.onNodeWithTag(PROMOTED_STORY_LIST_TEST_TAG).apply {
+      onChildAt(0)
+        .assertTextContains("Prototype Exploration")
+        .assertTextContains("FIRST TEST TOPIC")
+        .assertTextContains("SCIENCE")
+        .assertIsDisplayed()
+
+      onChildAt(1)
+        .assertTextContains("What is a Ratio?")
+        .assertTextContains("RATIOS AND PROPORTIONAL REASONING")
+        .assertTextContains("MATHS")
+        .assertIsDisplayed()
+    }
+  }
+
+  @Test
+  fun testFragment_markCompletedRatiosStory0_recommendsFractions() {
+    fakeOppiaClock.setFakeTimeMode(FakeOppiaClock.FakeTimeMode.MODE_UPTIME_MILLIS)
+    storyProgressTestHelper.markCompletedRatiosStory0(
+      profileId = profileId,
+      timestampOlderThanOneWeek = false
+    )
+    logIntoAdminTwice()
+
+    composeRule.onNodeWithTag(PROMOTED_STORY_LIST_HEADER_TEST_TAG).onChildAt(0)
+      .assertTextContains(context.getString(R.string.recommended_stories))
+      .assertIsDisplayed()
+
+    composeRule.onNodeWithTag(PROMOTED_STORY_LIST_TEST_TAG).onChildAt(0)
+      .assertTextContains("What is a Fraction?")
+      .assertTextContains("FRACTIONS")
+      .assertTextContains("MATHS")
+      .assertIsDisplayed()
+  }
+
+  @Test
+  fun testFragment_noTopicProgress_initialRecommendationFractionsAndRatiosIsCorrect() {
+    logIntoAdminTwice()
+
+    composeRule.onNodeWithTag(PROMOTED_STORY_LIST_HEADER_TEST_TAG).onChildAt(0)
+      .assertTextContains(context.getString(R.string.recommended_stories))
+      .assertIsDisplayed()
+
+    composeRule.onNodeWithTag(PROMOTED_STORY_LIST_TEST_TAG).apply {
+      onChildAt(0)
+        .assertTextContains("What is a Fraction?")
+        .assertTextContains("FRACTIONS")
+        .assertTextContains("MATHS")
+        .assertIsDisplayed()
+
+      onChildAt(1)
+        .assertTextContains("What is a Ratio?")
+        .assertTextContains("RATIOS AND PROPORTIONAL REASONING")
+        .assertTextContains("MATHS")
+        .assertIsDisplayed()
+    }
+  }
+
+  @Test
+  fun testFragment_forPromotedActivityList_hideViewAll() {
+    fakeOppiaClock.setFakeTimeMode(FakeOppiaClock.FakeTimeMode.MODE_UPTIME_MILLIS)
+    storyProgressTestHelper.markInProgressSavedFractionsStory0Exp0(
+      profileId = profileId,
+      timestampOlderThanOneWeek = false
+    )
+    logIntoAdminTwice()
+
+    composeRule.onNodeWithTag(PROMOTED_STORY_LIST_HEADER_TEST_TAG).onChildAt(1)
+      .assertDoesNotExist()
+  }
+
+  @Test
+  @Ignore("Temporarily ignored as the test is failing.")
```

Comment:
What's the failure here?

Reply:
The promoted story header is `Stories For You` instead of `Recommended Stories` and also the asserted promoted story is not present in the list.

Added a TODO to get it fixed in the recommendation logic PR.